### PR TITLE
Default and derived variants in schema

### PR DESCRIPTION
**Pull request by mvahowe**
This PR refactors the schema to implement one approach to supporting multiple "shapes" of schema, specifically for default/derived variants. The next step would be to use a similar approach in the type section to support multiple flavorTypes and flavors.

The top-level schema is a oneOf, which combines two root-level objects from which the details have been moved to separate files. This reduces duplication while also providing a very explicit and easy to edit answer to "what does the top level look like?"

The variant field has been moved to the meta section, since whichever section it lives in needs to be duplicated. I expect the default and derived meta schemas to diverge significantly anyway.

### CODE REVIEW

File: schema/default_meta.schema.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "$id": "https://burrito.bible/schema/default_meta.schema.json",
+    "title": "Scripture Burrito Meta (Default)",
+    "type": "object",
+    "description": "Information about the Scripture Burrito metadata file (default variant).",
+    "properties": {
+        "variant": {
+            "type": "string",
+            "pattern": "^default$",
```
Comment (rdb) [inline schema/default_meta.schema.json]: Let's make this `"const": "default"` instead.  I know you're a big fan of regexes, but this one doesn't need it. ;-)

File: schema/default_meta.schema.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        }
+    },
+    "required": [
+        "version",
+        "variant",
+        "dateCreated",
+        "defaultLanguage"
+    ],
+    "additionalProperties": false
+}
```
Comment (rdb) [inline schema/default_meta.schema.json]: Nitpick: newlines at ends of files.

File: schema/config.schema.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (rdb) [inline schema/config.schema.json]: This file looks like it doesn't belong here.

File: schema/default_meta.schema.json (hunk 1 above)
Comment (mvahowe) [reply schema/default_meta.schema.json]: I was allowing for future expansion in case we find other spellings like dephault.

**Comment by mvahowe** [comment]
@rdb and I agreed to proceed with this schema strategy for now, and to consider refactoring once we have the basic functionality in place.

**Pull request merged**